cmd/security-agent/subcommands/check/command.go

```diff
@@ -106,24 +107,16 @@ func commandsWrapped(bundleParamsFactory func() core.BundleParams) []*cobra.Comm
 }
 
 // RunCheck runs a check
-func RunCheck(log log.Component, config config.Component, checkArgs *CliParams) error {
+func RunCheck(log log.Component, config config.Component, statsd statsd.Component, checkArgs *CliParams) error {
 	hname, err := hostname.Get(context.TODO())
 	if err != nil {
 		return err
 	}
 
-	var statsdClient *ddgostatsd.Client
+	var statsdClient ddgostatsd.ClientInterface
 	metricsEnabled := config.GetBool("compliance_config.metrics.enabled")
 	if metricsEnabled {
-		// Create a statsd Client
-		statsdAddr := os.Getenv("STATSD_URL")
-		if statsdAddr == "" {
-			// Retrieve statsd host and port from the datadog agent configuration file
-			statsdHost := pkgconfig.GetBindHost()
-			statsdPort := config.GetInt("dogstatsd_port")
-			statsdAddr = fmt.Sprintf("%s:%d", statsdHost, statsdPort)
-		}
-		cl, err := ddgostatsd.New(statsdAddr)
+		cl, err := statsd.GetForHostPort(pkgconfig.GetBindHost(), config.GetInt("dogstatsd_port"))
```

Comment:
Is there a technical reason we want to create a new client for every `RunCheck`? 
Only asking because logically it makes more sense from a component perspective to create a single instance of the client within the component and share it as needed (vs having the component create new clients each time you call `Get...`)

Reply:
We should not create an instance of the statsd client into the component because there are valid reasons to want different client with different settings. `RunCheck` in particular could be keeping its client, but I didn't want to change that

Reply:
I'm not changing the semantic of RunCheck because I'm not sure why it was done this way in the first place. That being said it makes sense to have different client per check because it will provide some isolation between checks